### PR TITLE
Update shellcheck version

### PR DESCRIPTION
**Pull request by Vaibhav-Nazare**
Fix for report_too.sh 
Expanding an array without an index only gives the element in the index 0.
```
./.azure/scripts/shellcheck.sh

In tools/report.sh line 34:
if [[ "$BASH_VERSINFO" -lt 4 ]]; then
^------------^ SC2128: Expanding an array without an index only gives the first element.
```

### CODE REVIEW

**Comment by scholzj** [comment]
@Vaibhav-Nazare What shellcheck version are you using? Maybe you should sync it with the Strimzi CI which uses 0.7.2?

**Review by fvaleri — COMMENTED** [review]
Jakub is right, you are probably using shellcheck 0.7.1 or lower:

```sh
$ ./shellcheck-v0.7.1/shellcheck ~/Documents/strimzi-kafka-operator/tools/report.sh 

In /home/fvaleri/Documents/strimzi-kafka-operator/tools/report.sh line 34:
if [[ "$BASH_VERSINFO" -lt 4 ]]; then
       ^------------^ SC2128: Expanding an array without an index only gives the first element.

For more information:
  https://www.shellcheck.net/wiki/SC2128 -- Expanding an array without an ind...
```

**Comment by Vaibhav-Nazare** [comment]
I did check this at our end it's higer
```
shellcheck -V
ShellCheck - shell script analysis tool
version: 0.9.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

**Comment by scholzj** [comment]
Right. Then I think there are in general 2 options:
* You downgrade yours to 0.7.2 (instead of fixing the issue 0.9.0 reports). 
* Or you upgrade the one in our CI pipelines to 0.9.0 as well (plus fix the issue it reports). 

But running different versions and opening a PR each time it finds something different does not make sense. It might even happen that one day it has some conflicting issues. So bringing the versions in-sync one way or another will prevent this kind of issue from reappearing again and again.

I do not have a strong preference for which one to use - I guess updating the shellcheck version to 0.9.0 here might make more sense.

**Comment by Vaibhav-Nazare** [comment]
Yes upgrading it too 0.9.0 makes more sense

**Comment by fvaleri** [comment]
I also agree that we should upgrade shellcheck to 0.9.0 and fix the issue.

Note that the same check is also used in `docker-images/artifacts/build.sh` and `docker-images/kafka-based/build.sh`.

**Review by scholzj — APPROVED** [review]
Thanks for the changes!

File: tools/report.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ error() {
 if [[ -z ${BASH_VERSINFO+x} ]]; then
   error "No bash version information available, aborting"
 fi
-if [[ "$BASH_VERSINFO" -lt 4 ]]; then
+if [[ "${BASH_VERSINFO[*]}" -lt 4 ]]; then
```
Comment (fvaleri) [inline tools/report.sh:34]: This is not correct, as we only care about the major version, so index 0. The same in the other places.

```sh
$ if [[ "${BASH_VERSINFO[*]}" -lt 4 ]]; then
  error "You need bash version >= 4 to run the script"
fi
bash: [[: 5 2 15 1 release x86_64-redhat-linux-gnu: syntax error in expression (error token is "2 15 1 release x86_64-redhat-linux-gnu")
```

**Review by fvaleri — CHANGES_REQUESTED** [review]

**Review by ppatierno — APPROVED** [review]
LGTM.Thanks for the PR!

**Review by fvaleri — APPROVED** [review]
Thanks.

**Comment by scholzj** [comment]
Thanks for the PR

**Pull request merged**